rp2040-hal/src/pll.rs

```diff
@@ -50,6 +50,32 @@ pub struct PhaseLockedLoop<S: State, D: PhaseLockedLoopDevice> {
     state: S,
 }
 
+mod private {
+    use crate::pll::PhaseLockedLoopDevice;
+
+    pub trait PhaseLockedLoopReset: PhaseLockedLoopDevice {
+        fn bring_pll_up(&self, resets: &mut rp2040_pac::RESETS);
+    }
+}
+
+impl private::PhaseLockedLoopReset for rp2040_pac::PLL_SYS {
+    fn bring_pll_up(&self, resets: &mut rp2040_pac::RESETS) {
+        resets.reset.modify(|_, w| w.pll_sys().clear_bit());
+        while resets.reset_done.read().pll_sys().bit_is_clear() {
+            cortex_m::asm::delay(10);
```

Comment:
The delays inside these busy-wait loops don't add any value.
Having nothing in the body does the same thing but with no code cost, and will allow us to finish sooner.

Reply:
Was just copying how `gpio` handled the resets. I believe you are right that an empty body would be better.

Also, a note in case it comes up: I know I'm duplicating code here, but given there are only 2 options each I feel like the duplicated code is better than introducing a maco.